traits/observation/parsing.py

```diff
@@ -254,6 +254,7 @@ def _handle_tree(tree, default_notifies=None):
         tree.children, default_notifies=default_notifies)
 
 
+@lru_cache(maxsize=128)
```

Comment:
One (non-blocking) suggestion: it's probably worth pulling the `128` out into a (private) named constant, and defining that constant near the top of the file with a comment line explaining what it's for. E.g., something like:
```python
_OBSERVE_PATTERN_CACHE_MAXSIZE = 128
```
(but I haven't put any thought into a good name, so feel free to come up with something better)

Reply:
Hmm. It's really an expression cache, not a pattern cache. Given that it's a private variable and that the `observation` context should be clear, maybe `_EXPRESSION_CACHE_MAXSIZE`?

Reply:
I too prefer a module level constant (the truly desperate can change the value if they needed to).

Reply:
I agree with the module level constant, unfortunately since the cache is created via a decorator at import time (I think), there is no way to change it at run time by changing the constant - you'd need to patch the source.

Reply:
> there is no way to change it at run time by changing the constant - you'd need to patch the source.

Ah yes you are right. Nevermind that then, let's leave that to the truly desperate.

Reply:
yikes, I made this change locally but forgot to push... I will open a trivial separate PR adding the module level constant